pkg/util/cluster/aroclient.go

```diff
@@ -0,0 +1,116 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	v20231122 "github.com/Azure/ARO-RP/pkg/api/v20231122"
+	v20240812preview "github.com/Azure/ARO-RP/pkg/api/v20240812preview"
+	mgmtredhatopenshift20231122 "github.com/Azure/ARO-RP/pkg/client/services/redhatopenshift/mgmt/2023-11-22/redhatopenshift"
+	mgmtredhatopenshift20240812preview "github.com/Azure/ARO-RP/pkg/client/services/redhatopenshift/mgmt/2024-08-12-preview/redhatopenshift"
+	"github.com/Azure/ARO-RP/pkg/env"
+	redhatopenshift20231122 "github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/redhatopenshift/2023-11-22/redhatopenshift"
+	redhatopenshift20240812preview "github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/redhatopenshift/2024-08-12-preview/redhatopenshift"
+)
+
+type InternalClient interface {
+	Get(ctx context.Context, resourceGroupName string, resourceName string) (*api.OpenShiftCluster, error)
+	CreateOrUpdateAndWait(ctx context.Context, resourceGroupName string, resourceName string, parameters *api.OpenShiftCluster) error
+	DeleteAndWait(ctx context.Context, resourceGroupName string, resourceName string) error
+}
+
+type clientCluster interface {
+	mgmtredhatopenshift20231122.OpenShiftCluster | mgmtredhatopenshift20240812preview.OpenShiftCluster
+}
+
+type apiCluster interface {
+	v20231122.OpenShiftCluster | v20240812preview.OpenShiftCluster
+}
+
+type externalClient[ClientCluster clientCluster] interface {
+	Get(ctx context.Context, resourceGroupName string, resourceName string) (ClientCluster, error)
+	CreateOrUpdateAndWait(ctx context.Context, resourceGroupName string, resourceName string, parameters ClientCluster) error
+	DeleteAndWait(ctx context.Context, resourceGroupName string, resourceName string) error
+}
+
+type internalClient[ClientCluster clientCluster, ApiCluster apiCluster] struct {
```

Comment:
ApiCluster parameter is no longer necessary 
```suggestion
type internalClient[ClientCluster clientCluster] struct {
```

Reply:
nvm, this is needed in order to call `new(ApiCluster)`.